libs/platform/platform_tests_support/helpers.cpp

```diff
@@ -0,0 +1,43 @@
+#include "helpers.hpp"
+
+#include "base/assert.hpp"
+
+namespace platform::tests_support
+{
+time_t GetUnixtimeByDate(uint16_t year, Month month, uint8_t monthDay, uint8_t hours, uint8_t minutes)
+{
+  std::tm t{};
+  t.tm_year = year - 1900;
+  t.tm_mon = static_cast<int>(month) - 1;
+  t.tm_mday = monthDay;
+  t.tm_hour = hours;
+  t.tm_min = minutes;
+  return mktime(&t);
+}
+
+time_t GetUnixtimeByDate(uint16_t year, Month month, Weekday weekday, uint8_t hours, uint8_t minutes)
+{
+  int monthDay = 1;
+  auto createUnixtime = [&]()
+  {
+    std::tm t{};
+    t.tm_year = year - 1900;
+    t.tm_mon = static_cast<int>(month) - 1;
+    t.tm_mday = monthDay;
+    t.tm_wday = static_cast<int>(weekday) - 1;
+    t.tm_hour = hours;
+    t.tm_min = minutes;
+    return mktime(&t);
+  };
+
+  for (; monthDay < 32; ++monthDay)
+  {
+    auto const unixtime = createUnixtime();
+    auto const timeOut = localtime(&unixtime);
+    if (timeOut->tm_wday == static_cast<int>(weekday) - 1)
```

Comment:
If the intention is to return only the first day of the week of the month, then the function should be called differently, e.g. GetUnixtimeByFirstDayOfWeekInMonth